lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -27,9 +27,118 @@ class AwsCompileCloudFrontEvents {
       'before:remove:remove': this.logRemoveReminder.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8025
+    const originObjectSchema = {
+      type: 'object',
+      properties: {
+        ConnectionAttempts: { type: 'integer', miminum: 1, maximum: 3 },
+        ConnectionTimeout: { type: 'integer', miminum: 1, maximum: 10 },
+        CustomOriginConfig: {
+          type: 'object',
+          properties: {
+            HTTPPort: { type: 'integer' },
+            HTTPSPort: { type: 'integer' },
+            OriginKeepaliveTimeout: { type: 'integer', miminum: 1, maximum: 60 },
+            OriginProtocolPolicy: {
+              type: 'string',
+              enum: ['http-only', 'match-viewer', 'https-only'],
+            },
+            OriginReadTimeout: { type: 'integer', miminum: 1, maximum: 60 },
+            OriginSSLProtocols: {
+              type: 'array',
+              items: { type: 'string', enum: ['SSLv3', 'TLSv1', 'TLSv1.1', 'TLSv1.2'] },
+            },
+          },
+          additionalProperties: false,
+          required: ['OriginProtocolPolicy'],
+        },
+        DomainName: { type: 'string' },
+        Id: { type: 'string' },
+        OriginCustomHeaders: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              HeaderName: { type: 'string' },
+              HeaderValue: { type: 'string' },
+            },
+            additionalProperties: false,
+            required: ['HeaderName', 'HeaderValue'],
+          },
+        },
+        OriginPath: { type: 'string' },
+        S3OriginConfig: {
+          type: 'object',
+          properties: {
+            OriginAccessIdentity: {
+              type: 'string',
+              pattern: '^origin-access-identity/cloudfront/.+',
+            },
+          },
+          additionalProperties: false,
+        },
+      },
+      additionalProperties: false,
+      required: ['DomainName', 'Id'],
+      oneOf: [{ required: ['CustomOriginConfig'] }, { required: ['S3OriginConfig'] }],
+    };
+
+    const behaviorObjectSchema = {
+      type: 'object',
+      properties: {
+        AllowedMethods: {
+          oneOf: [
+            { type: 'array', items: [{ const: 'GET' }, { const: 'HEAD' }] },
+            { type: 'array', items: [{ const: 'GET' }, { const: 'HEAD' }, { const: 'OPTIONS' }] },
+            {
+              type: 'array',
+              items: [
+                { const: 'GET' },
+                { const: 'HEAD' },
+                { const: 'OPTIONS' },
+                { const: 'PUT' },
+                { const: 'PATCH' },
+                { const: 'POST' },
+                { const: 'DELETE' },
+              ],
+            },
```

Comment:
If understand we allow three exact collections. If it's the case then above doesn't meet that, as `AllowedMethods: ["GET"]` will pass.

I believe we can achieve needed validation through `{ enum: [["GET", "HEAD"], [... ], [...]] }`